web/src/user_topics.js

```diff
@@ -58,6 +63,54 @@ export function get_muted_topics() {
     return topics;
 }
 
+export function set_user_topic_visibility_policy(stream_id, topic, visibility_policy, from_hotkey) {
+    const data = {
+        stream_id,
+        topic,
+        visibility_policy,
+    };
+
+    channel.post({
+        url: "/json/user_topics",
+        data,
+        success() {
+            if (visibility_policy === all_visibility_policies.INHERIT) {
+                feedback_widget.dismiss();
+                return;
+            }
+            if (!from_hotkey) {
```

Comment:
I think the order should be:
```js
if (!from_hotkey) {
      return;
 }

if (visibility_policy === all_visibility_policies.INHERIT) {
     feedback_widget.dismiss();
     return;
}
 ```
 Because, if the request is not via hotkey, there is no point of `feedback.dismiss()` -- right?

Reply:
Sure, I will wait to address this along with the feedback from @timabbott.

Reply:
I think the code is actually better as written; the `feedback_widget` dismissal is to avoid continuing to show an inaccurate `feedback_widget` state where it's saying you just muted a topic, which you've in fact already changed to not be muted.

And there it doesn't matter whether you used the keyboard on a mouse button to unmute the topic; just whether the feedback widget is visible.